DashAI/back/routers/datasets.py

```diff
@@ -0,0 +1,112 @@
+import os
+
+import pydantic
+from fastapi import APIRouter, File, Form, UploadFile, status
+from fastapi.encoders import jsonable_encoder
+from fastapi.exceptions import HTTPException
+from sqlalchemy import exc
+
+from DashAI.back.database import db, models
+from DashAI.back.dataloaders.classes.csv_dataloader import CSVDataLoader
+
+# from Dataloaders.classes.audioDataLoader import AudioDataLoader
+# from Dataloaders.classes.csvDataLoader import CSVDataLoader
+# from Dataloaders.classes.imageDataLoader import ImageDataLoader
+from DashAI.back.dataloaders.classes.dataloader_params import DatasetParams
+from DashAI.back.models.classes.getters import get_model_params_from_task
+from DashAI.back.routers.session_class import session_info
+from DashAI.back.tasks.task import Task
+
+router = APIRouter()
+
+
+def parse_params(params):
+    """
+    Parse JSON from string to pydantic model
+    """
+    try:
+        model = DatasetParams.parse_raw(params)
+        return model
+    except pydantic.ValidationError as e:
+        raise HTTPException(
+            detail=jsonable_encoder(e.errors()),
+            status_code=status.HTTP_422_UNPROCESSABLE_ENTITY,
+        ) from e
+
+
+@router.get("/dataset/")
+async def get_dataset():
```

Comment:
What if I want only a specific dataset?

Reply:
Here I would leave the door open to implement a get with queries.
https://fastapi.tiangolo.com/tutorial/query-params/